### PR TITLE
Fix undefined.map error

### LINKED ISSUE #536: TypeError: undefined is not a function (near '...d.map...')
Sentry Issue: [APP-20-NEXT-5G](https://sentry.io/organizations/plant-for-the-planet/issues/2001577826/?referrer=github_integration)

```
TypeError: undefined is not a function (near '...d.map...')
  at c (./src/features/user/UserProfile/components/ProjectsContainer.tsx:89:23)
  at pi ([native code])
  at Po ([native code])
  at None (./src/features/user/UserProfile/components/ProjectsContainer.tsx:41:9)
  at promiseReactionJob ([native code])
...
(12 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by sarveshpro**
Fixes #536 

Changes in this pull request:
-  perform defined check before using map function on dynamic values.

### CODE REVIEW

File: src/features/projects/components/projectDetails/PlantSpecies.tsx [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
 
-function PlantSpecies({species}: Props): ReactElement {
-    return (
-        <div className={styles.projectMoreInfo}>
-            <div className={styles.infoTitle}>
-                {species.length} Species Planted
+function PlantSpecies({ species }: Props): ReactElement {
+  return (
+    <div className={styles.projectMoreInfo}>
+      <div className={styles.infoTitle}>{species.length} Species Planted</div>
```
Comment (sagararyal) [inline src/features/projects/components/projectDetails/PlantSpecies.tsx:15]: translate text

File: src/features/projects/components/projectDetails/ProjectInfo.tsx [hunk 2]
```diff
@@ -1,115 +1,137 @@
...
-    const calculatePercentage =(amount:any)=>{
-        const maxAmount = Math.max(...expenseAmount)
-        let percentage = (amount/maxAmount) * 100;
-        return `${percentage}%`
+      }
+      setOwnerTypes(newSiteOwners);
     }
+  }, []);
+
+  let expenseAmount = project.expenses.map((expense: any) => expense.amount);
```
Comment (norbertschuler) [inline src/features/projects/components/projectDetails/ProjectInfo.tsx:73]: This map function you have not added a safeguard for.

**Review by norbertschuler — APPROVED** [review]
With all the code reformatting the changes are very difficult to see. But I think it looks good.

**Comment by norbertschuler** [comment]
@sarveshpro please fix merge conflicts (merge develop or rebase)

**Comment by sarveshpro** [comment]
It is just too difficult, creating a new PR.

**Pull request closed without merging**